### PR TITLE
name_to_url default string & unicode tests

### PR DESCRIPTION
**Pull request by orangejenny**
support https://github.com/dimagi/commcare-hq/pull/8255

drop preceding/trailing hyphens, and add a prefix string if the generated string is all numbers/hyphens

@emord / @kaapstorm

### CODE REVIEW

**Comment by czue** [comment]
i don't speak regex very well so will let someone else merge, but :+1: to tests!

File: dimagi/utils/name_to_url.py [hunk 1]
```diff
@@ -1,4 +1,9 @@
 import re
 
-def name_to_url(name):
-    return re.sub(r'[^0-9a-z]+', '-', name.strip().lower())
+def name_to_url(name, default=""):
+    url = re.sub(r'[^0-9a-z]+', '-', name.strip().lower())
+    if re.search('^[0-9\-]*$', url) and default:
+        url = "{}-{}".format(default, url)
+    url = re.sub(r'^-+', '', url)
+    url = re.sub(r'-+$', '', url)
```
Comment (kaapstorm) [inline dimagi/utils/name_to_url.py]: Probably quicker to replace lines 7 & 8 with `url = url.strip('-')`:

    >>>  '---foo-bar---'.strip('-')
    'foo-bar'

Comment (kaapstorm) [inline dimagi/utils/name_to_url.py]: (That is a bit picky, though, maybe. I like this PR. Feel free to self-merge.)

Comment (orangejenny) [inline dimagi/utils/name_to_url.py]: i like that, it's cleaner. updated.

**Comment by orangejenny** [comment]
addressed Norman's feedback, self-merging

**Pull request merged**